src/platform/checksum.cpp

```diff
@@ -28,8 +28,26 @@ namespace platform {
         {
             f->ReadAll( data );
         }
+        std::string dataNoEol("");
```

Comment:
you were clear, it's not final
I was looking for a good solution for removing both kind of EOLs without too much memory usage and iterating several times, so your suggestion is valuable